test/generators/generator.test.js

```diff
@@ -304,4 +304,57 @@ describe('Generator', () => {
       });
     });
   });
+  describe('creating component and test files wrapped in a folder', () => {
+    const templatePath = '/tmp/components/Dumb.js';
+    const creationPath = '/components';
+    const componentName = 'Example';
+    const args = {
+      templatePath,
+      creationPath,
+      componentName,
+      settings: new MockSettings({
+        sourceBase: './tmp/src',
+        testBase: './tmp/test',
+        wrapFilesInFolders: true
+      })
+    };
+    const generator = new Generator(args);
+
+    beforeEach(() => {
+      sinon.spy(console, 'log');
+    });
+
+    afterEach(() => {
+      console.log.restore();
+    });
+
+    describe('#createComponent', () => {
+      it('renders template and writes it to componentPath', () => {
+        sinon.stub(generator, 'renderTemplate').returns('component file');
+
+        generator.createComponent();
+        const file = fs.readFileSync(generator.componentPath(), 'utf8');
+        expect(console.log.calledOnce).to.be.true;
+        expect(file).to.eq('component file');
+
+        fse.removeSync(generator.componentPath());
+        generator.renderTemplate.restore();
+      });
+    });
+
+    describe('#createTest', () => {
+      it('renders template and write it to testPath', () => {
+        sinon.stub(generator, 'renderTemplate').returns('test file');
+
+        generator.createTest();
+
+        const file = fs.readFileSync(generator.componentTestPath(), 'utf8');
+        expect(console.log.calledOnce).to.be.true;
+        expect(file).to.eq('test file');
+
+        fse.removeSync(generator.componentTestPath());
+        generator.renderTemplate.restore();
+      });
+    });
+  });
```

Comment:
Good point, working on the other tests now.

> Also test generated path: should it match the componentPath with the nested folder or when you keep your comps in folders do you not nest them in test directory as well?

I'm not really sure, I thought so for consistency, but maybe not?